### ISSUE TITLE
Working with Mojo behind API Gateway

### ISSUE THREAD

**Issue by pplu** [shogo82148/p5-aws-lambda#7]
Hi,

  I'm playing with AWS::Lambda::PSGI to serve a Mojo::Lite application. I'm currently building my Lambda layers with https://github.com/pplu/perl-lambda-byor, and using AWS::Lambda::PSGI to convert the Mojo::Lite app to something that Lambda can serve. The code for my Lambda function is:

```
use utf8;
use warnings;
use strict;
use AWS::Lambda::PSGI;
use JSON::MaybeXS;

$ENV{ PLACK_ENV } = 1;

my $app = require "$ENV{'LAMBDA_TASK_ROOT'}/lib/app.pl";
my $func = AWS::Lambda::PSGI->wrap($app);

sub main {
  my $payload = shift;
  return encode_json($func->($payload));
}

1;
```

This gets the application running, but there is one problem: links generated with `url_for` are not correct. When I generate a link for a controller, the link goes to "/controller", without putting the stage in front of the URL (the correct path should be "/prod/controller"). I've tracked the problem down to the way `AWS::Lambda::PSGI` builds the `SCRIPT_NAME` field of the PSGI env. Mojo generates correct links with:

```perl
    if (defined $payload->{ requestContext } and defined $payload->{ requestContext }->{ stage }) {
      $env->{SCRIPT_NAME} = "/$payload->{ requestContext }->{ stage }";
    } else {
      $env->{SCRIPT_NAME} = '';
    }
```

What do you think about this change?

**Comment by shogo82148** [comment]
Thank you for your feedback.

It looks good on `*.execute-api.${REGION}.amazonaws.com"` domains, but does it work on custom domains?
There is no stage name in the path if the api runs on custom domains.

**Comment by shogo82148** [comment]
How about `$env->{REQUEST_URI} = $payload->{ requestContext }->{ path }` 🤔 
It is the real path of the request.

**Comment by pplu** [comment]
I'm not sure it will work. I'll try it tonight. Here is the code that handles the incoming PSGI env in Mojo: https://metacpan.org/release/Mojolicious/source/lib/Mojo/Message/Request.pm#L159

**Comment by pplu** [comment]
The change to REQUEST_URI didn't work. I've created PR #8 with a test case.

**Comment by shogo82148** [comment]
Thanks!
I merged #8
